core/src/test/java/org/apache/druid/data/input/impl/HttpEntityTest.java

```diff
@@ -0,0 +1,113 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.impl;
+
+import com.fasterxml.jackson.databind.InjectableValues.Std;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.net.HttpHeaders;
+import org.apache.commons.io.FileUtils;
+import org.apache.druid.data.input.InputSource;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.metadata.DefaultPasswordProvider;
+import org.apache.druid.metadata.PasswordProvider;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class HttpEntityTest
+{
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Test
+  public void testOpenInputStream() throws IOException, URISyntaxException
+  {
+    URI url = new URI("https://druid.apache.org/data/wikipedia.json.gz");
+    PasswordProvider passwordProvider = new DefaultPasswordProvider("");
+    final InputStream inputStream = HttpEntity.openInputStream(url, "", passwordProvider, 0);
+    File file = new File("resources/wikipedia_fetched.json.gz");
+    File file1 = new File("resources/wikipedia.json.gz");
+    FileUtils.copyInputStreamToFile(inputStream, file);
+    Assert.assertTrue(FileUtils.contentEquals(file, file1));
+  }
+  @Test
+  public void testPartialOpenInputStream() throws IOException, URISyntaxException
+  {
+    URI url = new URI("https://druid.apache.org/data/wikipedia.json.gz");
+    PasswordProvider passwordProvider = new DefaultPasswordProvider("");
+    final InputStream inputStream = HttpEntity.openInputStream(url, "", passwordProvider, 15);
+    File file = new File("resources/wikipedia_fetched.json.gz");
+    File file1 = new File("resources/wikipedia.json.gz");
+    FileUtils.copyInputStreamToFile(inputStream, file);
+    Assert.assertFalse(FileUtils.contentEquals(file, file1));
+  }
+
+  long offset = 15;
+  URI uri = Mockito.mock(URI.class);
+  URL url = Mockito.mock(URL.class);
+  URLConnection urlConnection = Mockito.mock(URLConnection.class);
+  InputStream inputStreamMock = Mockito.mock(InputStream.class);
+  String contentRange = StringUtils.format("bytes %d-%d/%d", offset, 1000, 1000);
```

Comment:
this should go inside a setup method or at the top of the class where all variables are declared.

Reply:
Alright, I'll change accordingly.